docs/contributing-guide.adoc

```diff
@@ -1,52 +1,121 @@
-= Contributing guide
+= Contributing to AeroGear JavaScript SDK
 
-== Local setup
+First, ensure you have the https://docs.npmjs.com/[latest npm] installed.
 
-1. Install required dependencies
+Then, to set everything up run in your terminal:
 [source,bash]
-----
 npm install
 npm run bootstrap
-----
-2. Compile code
-[source,bash]
-----
-npm run build
-----
-3. Run tests and lint
+
+== Project Structure
+The SDK is made by a group of modules, each one implemented in a different NPM package and all placed under `/packages`.
+
+Each module provides an API to use a particular AeroGear service.
+
+A main package called `core` contains a shared set of interfaces, classes and other functionalities that the rest of modules depend on.
+
+== Working on modules
+
+Modules are written in https://www.typescriptlang.org/[TypeScript] and tested by https://mochajs.org/[mocha].
+
+Once the development is done, the module is built and the source code is transpiled into a `dist/` folder and published. Typings are automatically generated during this step as well.
+
+== Commands
+
+Run this commands from the repo's root directory.
+
+==== Run tests with coverage
 [source,bash]
-----
 npm run test
-npm run lint
-----
 
-=== Using core in projects
+==== Run linter
+[source,bash]
+npm run lint
```

Comment:
this might be missing being surrounded with `----\n`, not sure if asciidoc supports this syntax for a single source line?

Reply:
I've been using https://asciidoclive.com so I think syntax and format-wise there shouldn't be any problems